source/dub/generators/build.d

```diff
@@ -248,6 +244,18 @@ class BuildGenerator : ProjectGenerator {
 		return false;
 	}
 
+	private NativePath getBuildCacheDirectory(GeneratorSettings settings, in Package pack, string build_id)
+	{
+		NativePath base_path;
+		if (settings.buildCacheDirectory != NativePath.init)
+			base_path = settings.buildCacheDirectory;
+		else if (settings.tempBuild)
+			base_path = getTempDir() ~ format("dub/builds");
+		else return pack.path ~ format(".dub/build/%s/", build_id);
+
+		return base_path ~ format("%s/%s/%s/%s/", pack.basePackage.name, pack.version_, build_id);
```

Comment:
Let's better use `buildPath` for all of these formats here.
This will also avoid the error on Windows:

```
std.format.FormatException@C:\dmd2\windows\bin\..\..\src\phobos\std\format.d(486): Orphan format specifier: %s
----------------
0x00007FF6305A94E3 in std.exception.bailOut!(std.format.FormatException).bailOut at C:\dmd2\windows\bin\..\..\src\phobos\std\exception.d(515)
0x00007FF6305A93E9 in std.exception.enforce!(std.format.FormatException).enforce!bool.enforce at C:\dmd2\windows\bin\..\..\src\phobos\std\exception.d(436)
0x00007FF63062B242 in std.format.formattedWrite!(std.array.Appender!string, char, string, dub.dependency.Version, string).formattedWrite at C:\dmd2\windows\bin\..\..\src\phobos\std\format.d(487)
0x00007FF6306E39AA in std.format.format!(char, string, dub.dependency.Version, string).format at C:\dmd2\windows\bin\..\..\src\phobos\std\format.d(6385)
0x00007FF63083BE36 in dub.generators.build.BuildGenerator.getBuildCacheDirectory at C:\projects\dub\source\dub\generators\build.d(256)
```

Reply:
I used `NativePath`'s concatenation operator now. It's actually strange that I didn't get the orphan format specifier error, since I wrote this on a Windows machine.